python/lsst/daf/butler/cli/butler.py

```diff
@@ -79,36 +90,80 @@ def _importPlugin(pluginName):
             return None
 
     @staticmethod
-    def _getLocalCommand(commandName):
-        """Search the commands provided by this package and return the command
-        if found.
+    def _mergeCommandLists(a, b):
+        """Combine two dicts whose keys are strings (command name) and values
+        are list of string (the package(s) that provide the named command).
 
         Parameters
         ----------
-        commandName : string
-            The name of the command to search for.
+        a : dict of string to list of string
+            key is command name, value is packages that provide the command
+        b : (same as a)
 
         Returns
         -------
-        Command object
-            A Click command that can be executed.
+        dict of string to list of string
+            returns a extended with b
         """
-        try:
-            return getattr(butlerCommands, commandName)
-        except AttributeError:
-            pass
-        return None
+        for key, val in b.items():
+            try:
+                a[key].extend(val)
+            except KeyError:
+                a[key] = val
+        return a
 
     @staticmethod
     def _getLocalCommands():
         """Get the commands offered by daf_butler.
 
         Returns
         -------
-        list of string
-            The names of the commands.
+        dict of command name to list of packages name.
+            Key is the command name, value is the package that contains the
+            command.
         """
-        return [funcNameToCmdName(f) for f in butlerCommands.__all__]
+        return {funcNameToCmdName(f): ["lsst.daf.butler.cli.cmd"] for f in butlerCommands.__all__}
+
+    @classmethod
+    def _getPluginCommands(cls):
+        """Get the commands offered by plugin packages.
+
+        Returns
+        -------
+        dict of command name to list of packages name.
+            Key is the command name, value is the package(s) that contains the
+            command.
+        """
+        commands = {}
+        for pluginName in cls._getPluginList():
+            plugin = cls._importPlugin(pluginName)
```

Comment:
We have to keep an eye out for import overhead here. There is a worry that `butler create` could end up importing multiple obs and pipe_ packages in their entirety, which would be slow.